\documentclass{article}
\usepackage{graphicx}
\usepackage[letterpaper, portrait, margin=1in]{geometry}
\graphicspath{ {/Users/saralinker/Documents/SalkProjects/ME/GONetwork/GN_fig/} }
\title{GONetwork}
\author{Sara B. Linker, Sarah Voter, Fred H Gage}

\usepackage{Sweave}
\begin{document}
\maketitle
\SweaveOpts{concordance=TRUE}

\section*{Introduction}
GONetwork is an R toolkit that detects statistically significant functional annotation differences between gene lists and has direct application to RNA-seq, ChIP-seq, and GWAS experiments. Through a short workflow GONetwork: (1) calculates a gene network based on functional overlap, (2) clusters genes within that defined functional space, and (3) statistically tests differential enrichment within each cluster to determine functional annotation that is unique to a given condition.


 \section*{Background}
Currently, standard functional analyses are restricted by the requirement of exact overlap of a gene list with a dictionary of predefined terms. This strategy is limited by the lack of complete annotation. Furthermore, it is limited by reduced power in small genes sets which often results in broad, biologically ambiguous terms. Contrary to these current methods that require functional equivalence, GONetwork compares genes via functional similarity. Functional similarity initially broadens the space of terms identified and then narrows in on the highly specific terms within a gene set. This alternative strategy returns a highly robust and informative set of enriched functional terms. Within the terms that are enriched within a complete dataset, GONetwork then tests differential enrichment of each annotation cluster based on the condition from which the gene was identified. GONetwork returns to the user a highly specific set of functional terms that are relevant to the study as well as those that are specific to a given condition. This enables researchers to narrow in on the informative functional groupings present within their dataset.

<<Internal Functions, echo=FALSE>>=
library(GONetwork)
@

\section*{Installation}
<<Installation, echo=T>>=
library(devtools)
#install_github("saralinker/GONetwork")
@

\section*{Run}
GONetwork is designed to take a list of genes and calculate a similarity matrix based on functional annotation. The matrix (M) is created with with knowledge of the hierarchical representation of GO terms. Since there are many GO terms available and few GO terms relevant for each gene, M is a highly sparse matrix. A cosine function is applied to successfully work with this sparse data, however the user can input other common distance measures. We've provided a test dataset to work with generated from real data examining neural progenitor cells throughout differentiation into neurons. Gene expression was compared between neural progenitor cells and neurons. \textsl{differentiation} contains the significant genes from this analysis.

<<Load Data, echo=T>>=
library(GONetwork)
data(differentiation)
head(differentiation)
@

\subsection*{Generate GOterm Matrix}
Use \textbf{getGo()} to create the matrix of GO terms based on the gene list.
<<getGo, echo=T>>=
M <- getGo(differentiation,species = "human")
@


Once the GO term matrix is created, use \textbf{GoTheDist()} to calculate the distances between genes. Here, we're using the cosine distance to calculate similarity. It is useful to restrict the analysis to highly informative GO terms. Here we've required that all of the GO terms have at least 15 ancestors. 
\vspace{5mm}

<<GoTheDist, echo=T>>=
D <- GoTheDist(M,method="cosine",minparents=15)
@

\subsection*{Plot the full network}
\textbf{cyto()} will convert the matrix to Cytoscape format. The overall connectivity of the network can be adjusted by changing the cutoff values. Here, we exclude genes with a similarity score of less that 0.2.
\vspace{5mm}
<<cyto,echo=T>>=
tab<-cyto(D,cutoff = 0.3)
@
\begin{figure}[ht]
\includegraphics[width=12cm]{distance_clustering_all.png}
\centering
\caption{Cytoscape plot of all genes labelled by gene cluster}
\end{figure}

\subsection*{Identify gene clusters}
\textbf{AssignCluster()} will determine gene clusters given a user defined k. Here we've turned on the additional option to remove stragglers. This recalculates gene clusters after removing genes that are not clearly defined with GO annotation.
\vspace{5mm}
<<AssignCluster,echo=T>>=
tab <-AssignCluster(tab = tab, D = D, K = 30, rmv_stragglers = TRUE)
@
\begin{figure}[ht]
\includegraphics[width=12cm]{distance_clustering_terms.png}
\centering
\caption{Cytoscape plot after refinement}
\end{figure}

\subsection*{Differential functional enrichment}
You can then calculate differential functional enrichment between conditions using \textbf{testClusters()}. Test clusters requires an additional variable \textbf{group} which identifies which condition each gene is related to. \textbf{group} is a data.frame where the first column contains all of the gene names from \textbf{differentiation} and the second column contains the group designation to be tested ("up-regulated" or "down-regulated").

\begin{figure}[ht]
\includegraphics[width=12cm]{group_clustering_v2.png}
\centering
\caption{Genes grouped by clustering and colored by differential expression estimate. All clusters represent genes that are enriched for specific functional annotation. Note the interconnectedness wihtin a group as well as between groups. The between groups represents gene clusters with similar annotation. Highlighted clusters contain more genes from a given condition. For example, cluster 22 primarily contains genes that are more highly expressed in neurons. Below we will examine what genes and terms are present within these clusters. }
\end{figure}

<<testClusters,echo=T>>=
data(group)
k.test <- testClusters(tab,group)
k.test[k.test$prop_test_p < 0.05,]
@

if you would like to make this group dataframe from your own data here is an example using results from the edgeR differential expression outout. Here we are labelling genes as "high" if the logFC is greater than 0 and "low" if less than 0.
<<make group,echo=T>>=
data(res)
group <- data.frame(genes = differentiation,
                    group = ifelse(res[differentiation, "logFC"] > 0, "high","low"))
@

\subsection*{Examine annotation associated with each group}
\textbf{findTerms} returns a list of GO terms shared within each gene group. Here, we've required that at least 40 percent of the genes share the annotation term.
<<findTerms,echo=T>>=
term_genes <- unique(as.character(tab[as.character(tab$group) == 4, "origin"]))
head(term_genes)
findTerms(M, term_genes,proportion.shared = 0.5)
findTerms(M, term_genes,proportion.shared = 0.2)
@

\subsection*{Examine genes associated with each annotation}
\textbf{findGenes} returns genes in your dataset exactly associated with a given functional term.
<<findGenes,echo=T>>=
findGenes(M,tab, group = 21, term = "DNA repair")
@

Alternatively, you can use \textbf{findGenes} to return genes in any GO term. Here we are looking for which genes, and associated clusters, contain genes involved in the cell cycle.
<<findGenes,echo=T>>=
fg <- findGenes(M,tab, term = "postsynaptic membrane")
head(fg)
@

We can then trace back which are the top clusters that have this term. The term appears across many genes including the group 0, which are unclustered genes, and is enriched in group 21 which is a cluster that is significantly associated with genes that are up-regulated in neurons.
<<findGenes table,echo=T>>=
fg.table <- data.frame(table(fg$group))
fg.table <- fg.table[order(fg.table$Freq,decreasing = TRUE),]
head(fg.table)
@

Let's take a look at what the top functional categories are in 21, which contains postsynaptic membrane genes.
<<findTerms2,echo=T>>=
#Group 4
term_genes <- unique(as.character(tab[as.character(tab$group) == 21, "origin"]))
head(term_genes)
findTerms(M, term_genes,proportion.shared = 0.3)



\end{document}
